src/mpc-hc/MainFrm.cpp

```diff
@@ -17629,6 +17670,17 @@ void CMainFrame::CloseMedia(bool bNextIsQueued/* = false*/)
         }
     }
 
+    // save external subtitle
+    if (g_bExternalSubtitle &&
+        m_pCurrentSubInput.pSubStream && m_pCurrentSubInput.pSubStream->GetPath().IsEmpty()) {
+        const auto& s = AfxGetAppSettings();
+        if (s.bAutoSaveDownloadedSubtitles) {
+            int pos = 0;

```

Comment:
```
if (s.strSubtitlePaths.IsEmpty() || s.strSubtitlePaths.Left(2) == L".;") {
    SubtitlesSave(nullptr, true);
} else {
    const CString& dir = s.strSubtitlePaths.Tokenize(_T(";"), 0);
    SubtitlesSave(dir, true);
}
```

Reply:
Is it intentional that the `strSubtitlePaths == L"."` and the `s.strSubtitlePaths.Left(2) == L".;"` should be handled separately?

`Tokenize` requires a reference to a non-const int, so a literal cannot be passed to it.
https://docs.microsoft.com/en-us/cpp/atl-mfc-shared/reference/cstringt-class?view=msvc-170#tokenize

Reply:
This is to prevent passing an empty string for `dir`.
The comparison against `L".;"` is to avoid the unneeded code path for redirection in case of the default value of `strSubtitlePaths`. `L"."` is not a default value, so that does not really need a quick code path.